sw/device/lib/dif/dif_pinmux.h

```diff
@@ -336,8 +278,7 @@ typedef enum dif_pinmux_result {
  * @return The result of the operation.
  */
 OT_WARN_UNUSED_RESULT
-dif_pinmux_result_t dif_pinmux_init(dif_pinmux_params_t params,
-                                    dif_pinmux_t *pinmux);
+dif_result_t dif_pinmux_init(mmio_region_t base_addr, dif_pinmux_t *pinmux);
```

Comment:
Pinmux has `templated` attribute set, so its init DIF needs to take additional `dif_pinmux_params_t` arg so that it has a more standardized look. 

OR autogen the params and integrate into the context struct as I had suggested in the issue.

Reply:
(As we briefly discussed offline:) Yes, while the pinmux is technically templated, it's DIFs never had any extra bits of information in the params struct, besides the base address, so this was deprecated to be consistent with other IPs that also used no extra params. See #8409 for more details, specifically my comment [here](https://github.com/lowRISC/opentitan/issues/8409#issuecomment-931816697).

Reply:
Agreed (we discussed this offline).